### ISSUE TITLE
Investigate flood of `suggestionsDatabaseUpdate` messages on startup 

### ISSUE THREAD

**Issue by hubertp** [enso-org/enso#6706]
_Separated the issue from ticket https://github.com/enso-org/enso/issues/6630._

On a relatively modest project, on startup, we appear to send hundreds of small messages from language server with suggestions updates. ~This is intentional, to some extent, because we abandoned batch updates, but~ Apparently it can put a strain on IDE to handle them, especially within such a short period of time.

**Comment by wdanilo** [comment]
What is the reason batch updates were "abandoned"? If they were abandoned, there has to be reason for it. Instead of fixing this issue now somehow, we should chat about what was the reason and what is the proper solution here.

For example, if batch updates were too big to sent via network, flooding GUI with a lot of updates might be a good idea, but we should then implement their batching on our side. Anyway, this needs discussion/proper information before implementation.

**Comment by hubertp** [comment]
> What is the reason batch updates were "abandoned"? If they were abandoned, there has to be reason for it. Instead of fixing this issue now somehow, we should chat about what was the reason and what is the proper solution here.

DB updates for libraries suggestions were previously created during the startup. There were a lot of them and they were batched. Now we don't batch them but actually create a single message per library. So that was a typo on my side. 

The flooding that happens appears to be related to regular updates as things are processed. I'm saying "appears to be" because that's part of the investigation and part of the process. That's why I don't see the reason for "chatting about it" with GUI folks, yet. Maybe it's a problem on LS maybe GUI but unless we are actually given time to investigate I can't tell.

I can sense the urgency in your tone but I also don't intend to involve too many folks, too early, for obvious resourcing reasons.

**Comment by wdanilo** [comment]
Makes sense, thanks for the info. Regarding chatting – let's do it when we will have enough information, of course. If the problem with GUI is because it is being re-rendered a lot of times after these updates, it is not necessarily problem with the amount of updates – maybe we should just batch incoming updates in GUI and apply then next frame / after few frames. Sending updates as a batch might cause network issues and performance drop there (it might not, batched updates might be sent faster). Anyway, all these aspects need to be taken into consideration when we will have the information after the investigation :) 

Regarding "urgency" in my tone – that was definitely not intentional. The only thing I wanted to catch early here is that the fix for the issue is not obvious and fixes on GUI side might be needed as much as on the engine side. So before we get to implementation, I'd like to have a sync call between all parts here.

**Comment by hubertp** [comment]
Followed the hunch and it appears that we issue multiple suggestions updates even if the return type of expressions does not change. This is apparently especially painful if the graph has errors or panics. For the sake of the argument I turned those off and GUI became very responsive.
After discussing this with @4e6, it looks like this condition was changed in https://github.com/enso-org/enso/pull/3729, so I will try to resurrect the check and not break the "pending nodes" functionality. 

This sounds vaguely similar to https://github.com/enso-org/enso/issues/6639.

So far it looks like an issue on LS-side.
